tests/test_other.py

```diff
@@ -10580,3 +10580,7 @@ def test_report_subprocess_signals(self):
 
   def test_concepts(self):
     self.do_runf(test_file('other', 'test_concepts.cpp'), '', emcc_args=['-std=c++20'])
+
+  def test_link_only_setting_warning(self):
+    err = self.run_process([EMCC, '-sALLOW_MEMORY_GROWTH', '-c', test_file('hello_world.c')], stderr=PIPE).stderr
+    self.assertContained("warning: linker setting ignored during compilation: 'ALLOW_MEMORY_GROWTH' [-Wunused-command-line-argument]", err)
```

Comment:
Please add testing of the warning not showing up when the flag is passed to ignore it.

Reply:
Sure.. although in general we shouldn't need to that for all warnings since the system for enable/disabling is fully generic.

Reply:
Yes, maybe you're right and it is excessive if we have good testing elsewhere. lgtm without it.